controllers/lvm_volumegroup.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	lvmVGName = "lvmvg-manager"
+)
+
+type lvmVG struct{}
+
+func (c lvmVG) getName() string {
+	return lvmVGName
+}
+
+func (c lvmVG) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	lvmVolumeGroups := c.getLvmVolumeGroups(r, lvmCluster)
+
+	for _, volumeGroup := range lvmVolumeGroups {
```

Comment:
Creating the LVMVolumeGroup CRs from the LVMCluster CR is still a design concern. The admin should create the LVMVolumeGroup CRs directly. Not a blocker for feature complete, but we really need to discuss this.

Reply:
I disagree. The admin should create the LVMCluster CR and the controller should create the LVMVolumeGroup CRs. The intention is to have a single CR for the controller to create the internal pieces from.

Reply:
I still haven't seen a design backing this decision, so my position remains to disagree with it so we can follow standard K8s patterns.